File: wss_demo/python/ws_private_demo.py
import hmac
import json
import logging
import time

import websocket

logging.basicConfig(filename='logfile_wrapper.log', level=logging.DEBUG,
                    format='%(asctime)s %(levelname)s %(message)s')
topic = "order"

trade_results = {}
size = {}
size_usdt = {}


def on_message(ws, message):
    data = json.loads(message)
    print(data)


def on_error(ws, error):
    print('we got error')
    print(error)


def on_close(ws):
    print("### about to close please don't close ###")


def send_auth(ws):
    key = 'xxxx'
    secret = 'xxxx'
    expires = int((time.time() + 10) * 1000)
    _val = f'GET/realtime{expires}'
    print(_val)
    signature = str(hmac.new(
        bytes(secret, 'utf-8'),
        bytes(_val, 'utf-8'), digestmod='sha256'
    ).hexdigest())
    ws.send(json.dumps({"op": "auth", "args": [key, expires, signature]}))


def on_pong(ws, *data):
    print('pong received')


def on_ping(ws, *data):
    #now = datetime.now()
    #dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
    #print("date and time =", dt_string)
    print('ping received')


def on_open(ws):
    print('opened')
    send_auth(ws)
    print('send subscription ' + topic)
    ws.send(json.dumps({"op": "subscribe", "args": [topic]}))


def connWS():
    ws = websocket.WebSocketApp("wss://stream.bybit.com/realtime_private",
                                on_message=on_message,
                                on_error=on_error,
                                on_close=on_close,
                                on_ping=on_ping,
                                on_pong=on_pong,
                                on_open=on_open
                                )
    ws.run_forever(
        #http_proxy_host='127.0.0.1',
        #http_proxy_port=1087,
        ping_interval=30,
        ping_timeout=10
    )


if __name__ == "__main__":
    websocket.enableTrace(True)
    connWS()

File: wss_demo/js/ws_private_demo.js
var url = require('url');
var WebSocket = require('ws');
var crypto = require('crypto');
var endpoint = "wss://stream-testnet.bybit.com/realtime"
var parsed = url.parse(endpoint);
console.log('attempting to connect to WebSocket %j', endpoint);
var client = new WebSocket(endpoint);
const apiKey="U7OUiyDdJPVeU7fcxn";
const apiSecret="6o2s1eOnkViVLfyg3zii1k0kf56j5aBNkLRn";
client.on('open', function () {
	console.log('"open" event!');
	console.log('WebSocket Client Connected');
	const expires = new Date().getTime() + 10000;
	const signature = crypto.createHmac("sha256", apiSecret).update("GET/realtime" + expires).digest("hex");
	const payload={
		op: "auth",
		args: [apiKey, expires.toFixed(0), signature],
	}
	client.send(JSON.stringify(payload));
	setInterval(()=>{client.ping()}, 30000);
	client.ping();
	client.send(JSON.stringify({"op": "subscribe", "args": ['order']}));
});

client.on('message', function (data) {
	console.log('"message" event! %j', data);
});
client.on('ping', function (data, flags) {
	console.log("ping received");
});
client.on('pong', function (data, flags) {
	console.log("pong received");
});
